pkg/controller/scheduledjob/controller.go

```diff
@@ -0,0 +1,275 @@
+					// (concurrency/suspended rules will not apply in that case, obviously, since we can't
+					// stop users from creating jobs if they have permission.)  It is assumed that if a
+					// user has permission to create a job within a namespace, then they have permission to
+					// affect (possibly "trick") the parent.  ReplicaSets and their Pods work the same way.
+					ref, err := getRef(jobResp)
+					if err != nil {
+						glog.V(2).Infof("Unable to make object reference for job for %s", nameForLog)
+					} else {
+						sj.Status.Active = append(sj.Status.Active, *ref)
+						err := sjc.UpdateStatus(&sj)
+						if err != nil {
+							glog.Infof("Unable to update status for %s", nameForLog)
+							// Will notice and retry next sync.
+						} else {
+							recorder.Eventf(&sj, api.EventTypeNormal, "AdoptedJob", "Adopted job: %v", err)
+						}
+					}
+					// TODO: how to update sj.Status.LastScheduleTime if the adopted job is newer than any we knew about?
+			} else {
+				if idx := findInActiveList(sj.Status.Active, j.ObjectMeta.UID) ; idx != -1 {
+					deleteFromActiveList(&sj, idx)
+					// Nothing to do?
+					// TODO: event to call out failure vs success.
+					recorder.Eventf(&sj, api.EventTypeNormal, "SawCompletedJob", "Saw completed job: %v", err)
+				} else {
+					// Nothing to do
+				}
+			}
+		}
+		UpdateStatus(status)
+	*/
+
+	nameForLog := fmt.Sprintf("namespace/%s/scheduledJob/%s", sj.Namespace, sj.Name)
+
+	if sj.Spec.Suspend {
+		glog.V(4).Infof("Not starting job for %s because it is suspended", nameForLog)
+		return
+	}
+	times, err := getRecentUnmetScheduleTimes(sj, now)
+	if err != nil {
+		glog.Errorf("Cannot determine if %s needs to be started: %v", nameForLog, err)
+	}
+	// TODO: handle multiple unmet start times, from oldest to newest, updating status as needed.
+	if len(times) == 0 {
+		glog.V(4).Infof("No unmet start times for %s", nameForLog)
+		return
+	}
+	if len(times) > 1 {
+		glog.Errorf("Multiple unmet start times for %s so only starting last one", nameForLog)
+	}
+	scheduledTime := times[len(times)-1]
+	tooLate := false
+	if sj.Spec.StartingDeadlineSeconds != nil {
+		tooLate = scheduledTime.Add(time.Second * time.Duration(*sj.Spec.StartingDeadlineSeconds)).Before(now)
+	}
+	if tooLate {
+		glog.Errorf("Missed starting window for %s", nameForLog)
+		// TODO: generate an event for a miss.  Use a warning level event because it indicates a
+		// problem with the controller (restart or long queue), and is not expected by user either.
+		// Since we don't set LastScheduleTime when not scheduling, we are going to keep noticing
+		// the miss every cycle.  In order to avoid sending multiple events, and to avoid processing
+		// the sj again and again, we could set a Status.LastMissedTime when we notice a miss.
+		// Then, when we call getRecentUnmetScheduleTimes, we can take max(creationTimestamp,
+		// Status.LastScheduleTime, Status.LastMissedTime), and then so we won't generate
```

Comment:
I like this idea. Quite a lot of systems (builds, remote logins) provide you with an information about last failed attempt.